### PR TITLE
Mac os custom font loading fix

### PR DESCRIPTION
**Pull request by newnon**
Mac os user font from bundle loading fix
font load from path you don't even need to add it to plist file
second version of this patch https://github.com/cocos2d/cocos2d-x/pull/7578 - just truncate name internally as in iPhone to load font with path but you still need to add all fonts to .plist file
merge only one

### CODE REVIEW

**Comment by Mazyod** [comment]
Nice one!

I developed a hybrid iOS/Mac library in the past, and I use this to load all the fonts dynamically, so I don't need to add them to the plist and all that setup crap. They can easily be loaded using `fontWithName:`. Just throwing it out there:

``` objc
- (void)_loadAllFontsDynamically
{
    for (NSString *fontType in @[@"ttf", @"otf"])
    {
        NSArray *fonts = [[NSBundle lpBundle] recursivePathsForResourcesOfType:fontType];
        for (NSString *fontPath in fonts)
        {
            NSData *inData = [NSData dataWithContentsOfFile:fontPath];

            CFErrorRef error;
            CGDataProviderRef provider = CGDataProviderCreateWithCFData((CFDataRef)inData);
            CGFontRef font = CGFontCreateWithDataProvider(provider);

            if (!CTFontManagerRegisterGraphicsFont(font, &error))
            {
                CFStringRef errorDescription = CFErrorCopyDescription(error);
                NSLog(@"Failed to load font: %@", errorDescription);
                CFRelease(errorDescription);
            }

            DLog(@"Loaded font: %@", font);

            CFRelease(font);
            CFRelease(provider);
        }
    }
}
```

**Comment by newnon** [comment]
Nice, but cocos load font with full name on most platforms, thats why my method do exactly what we need.

**Comment by Mazyod** [comment]
Ah, well the more platform agnostic the better :)

**Comment by zilongshanren** [comment]
Thanks, @newnon  
Could you please add a test case in cpp-tests to stress this fix?

**Comment by newnon** [comment]
ok i'll try

**Comment by newnon** [comment]
i made second version of this patch https://github.com/cocos2d/cocos2d-x/pull/7578

**Comment by minggo** [comment]
So i can close this one?

**Comment by newnon** [comment]
you must decide which version of this path if more suitable for cocos. Both fix the same problem but with different solutions.

File: cocos/platform/ios/CCDevice.mm [hunk 1]
```diff
@@ -260,6 +260,8 @@ static bool _initWithString(const char * text, cocos2d::Device::TextAlign align,
         // the '.ttf' extensions when referring to custom fonts.
         fntName = [[fntName lastPathComponent] stringByDeletingPathExtension];
         
+        CTFontManagerCreateFontDescriptorsFromURL
```
Comment (Mazyod) [inline cocos/platform/ios/CCDevice.mm]: Isn't this causing a compilation error?

**Comment by minggo** [comment]
@newnon 
If i add fonts into plist, and merged this PR.
Then what will happen?

**Comment by newnon** [comment]
you already merged another one. after that you don't need any ifdef to load fonts just full path to it

**Pull request closed without merging**